test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -385,3 +385,21 @@
        (is (logged?
             #"shutdown-on-error triggered because of exception"
             :error))))))
+
+(deftest ^:integration overflow-test
+  (let [watch-dir (fs/temp-dir "overflow")
+        results (atom [])
+        callback (make-callback results)]
+    (with-app-with-config
+     app watch-service-and-deps {}
+     (let [service (tk-app/get-service app :FilesystemWatchService)]
+       (watch! service watch-dir callback))
+     ;; From experimenting the buffer seems to fill at 512 events.
+     ;; However the number of events varies by operating system.
+     (let [files (for [n (range 1000)]
```

Comment:
I think this can be more concisely written with `dotimes`.

```clojure
(dotimes [n 1000]
  (spit (format "%s/%s.txt" watch-dir n) "file content"))
```